File: import_to_firebase.py
import firebase_admin
from firebase_admin import credentials, firestore
import json
import os


# Initialize Firebase
cred = credentials.Certificate("<your-service-account-key>.json")
firebase_admin.initialize_app(cred)


db = firestore.client()


def import_collection(collection_name):
    with open(f'collections/{collection_name}.json', 'r') as file:
        data = json.load(file)
    for doc in data:
        db.collection(collection_name).document(doc['id']).set(doc)
        print(f"Document {doc['id']} added to collection {collection_name}")


collections = [
    'club',
    'divisions-scraping',
    'league',
    'leagues-scraping',
    'match',
    'matchstats',
    'player',
    'roles',
    'scraping-leagues',
    'team',
    'teams-scraping',
    'users'
]


for collection_name in collections:
    import_collection(collection_name)

File: node-scripts-equivalent/import_to_firebase.js
var admin = require("firebase-admin");
var fs = require('fs');
const path = require('path');
var serviceAccount = require("./<your-service-account-key>.json");


admin.initializeApp({
    credential: admin.credential.cert(serviceAccount)
});


var db = admin.firestore();


function importCollection(collectionName) {
    const data = JSON.parse(fs.readFileSync(`collections/${collectionName}.json`, 'utf8'));
    data.forEach(doc => {
        db.collection(collectionName).doc(doc.id).set(doc)
            .then(() => console.log(`Document ${doc.id} added to collection ${collectionName}`))
            .catch((error) => console.error(`Error adding document ${doc.id}:`, error));
    });
}


const collections = [
    'club',
    'divisions-scraping',
    'league',
    'leagues-scraping',
    'match',
    'matchstats',
    'player',
    'roles',
    'scraping-leagues',
    'team',
    'teams-scraping',
    'users'
];


collections.forEach(collectionName => {
    importCollection(collectionName);
});
